### PR TITLE
feat(wallet): add ability to remove wallet account

### LINKED ISSUE #18575: [Epic] -Feature- Wallet - Remove Account
All features related to issue for epic - https://github.com/status-im/status-mobile/labels/%3AMobile%20Wallet%20Feature%20-%20Delete%20Account

This issue should be hidden with a `feature-flag` so that it is not in the Nightly develop and only viewable/workable for those required to see the feature.

### User Story

- As a user, I want to be able to delete a wallet account.

- Depending on the account type the warning message should be different.
In particular generated accounts should warn me to appropriately back up keypair or recovery phrase and also the derivation path with an option to copy to this derivation path

- Once the wallet account is removed I want to instantly see this.

- This option should be able to be access from the account options actions and also from the edit page.

- Once an account is deleted a toast should show me that this activity has happened

- I should not be able to delete the first account I am given with my Status Profile and for this account the option to remove the account should not be visible.

- If I go to make a new wallet account, the default name for watch only address and account generation should be respective to how many wallet accounts I have visible in my Status Profile

- once I remove an account the balances should reflect the users updated balance.

### Tasks
Add UI to delete wallet account
- [x] https://github.com/status-im/status-mobile/issues/18421

Connect backend to delete wallet account, display toast and refresh accounts data
- [x] https://github.com/status-im/status-mobile/issues/18422

Hide Option to remove account for default wallet account
- [x] https://github.com/status-im/status-mobile/issues/18576

Default Account Names on creation are relative to how many wallet accounts the user has
- [x] https://github.com/status-im/status-mobile/issues/18416

### Designs 
[Figma Link](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=900%3A233949&mode=design&t=PjyYX6y911TQlcCl-1)

### Closing this issue
Once all of these issues are ready the corresponding feature Flag Can be removed. This pr to remove the feature flag can then be used for QA and Design to ensure everything is feature complete and this issue can be closed.

### PR DESCRIPTION
**Pull request by J-Son89**
closes: https://github.com/status-im/status-mobile/issues/18575

This pr removes the feature flag for removing an account - please see the attached issue for expected behaviour

### CODE REVIEW

**Review by briansztamfater — APPROVED** [review]

File: src/status_im/contexts/wallet/sheets/account_options/view.cljs [hunk 1]
```diff
@@ -100,12 +99,10 @@
...
-           :on-press            #(ff/alert ::ff/wallet.remove-account
-                                           (fn []
-                                             (rf/dispatch [:show-bottom-sheet
-                                                           {:content
-                                                            (fn []
-                                                              [remove-account/view])}])))})]]]
+           :on-press            #(rf/dispatch [:show-bottom-sheet
+                                               {:content
+                                                (fn []
+                                                  [remove-account/view])}])})]]]
```
Comment (smohamedjavid) [inline src/status_im/contexts/wallet/sheets/account_options/view.cljs]:
```suggestion
                                               {:content remove-account/view}])})]]]
```

**Review by smohamedjavid — APPROVED** [review]

**Review by OmarBasem — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
hi @J-Son89 thank you for the PR. Here is a list of the found issues. All of them are related to the UI, and I believe they can be fixed separately. Let me know if you are going to fix them within the scope of this PR. Otherwise, the PR can be merged, and I will create them separatly

## ISSUE 1: Missing Cancel button on the remove account drawer
### Steps:
1. Create an additional ETH account.
2. Open the account.
3. Open the remove drawer.

### Actual result:
The 'Cancel' button is not shown in the drawer.
image

### Expected result:
The 'Cancel' button should be shown.
image
https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=900-233949&mode=design&t=jsIpHlzMGJQUlOMh-0

**Comment by VolodLytvynenko** [comment]
## ISSUE 2: [Android] Defalut account is shifted after removing the last account available for removing

### Steps:
1. Create an additional ETH account.
2. Scroll the accounts to the left.
3. Remove the account.
4. Check the behavior of the last account.

### Actual result:
- After the account is removed, the default account is still shifted to the left.
- After scrolling the last account back, it is shifted to the right.

https://github.com/status-im/status-mobile/assets/52490791/259f6238-a589-4eaa-bc22-051cffcece15

### Expected result:
The account positions should be aligned
image

### Device:
Pixel 7a, Android 14

**Comment by J-Son89** [comment]
thanks @VolodLytvynenko - I will fix these issues in this pr 👍

**Comment by J-Son89** [comment]
> ## ISSUE 2: [Android] Defalut account is shifted after removing the last account available for removing
> ### Steps:
> 1. Create an additional ETH account.
> 2. Scroll the accounts to the left.
> 3. Remove the account.
> 4. Check the behavior of the last account.
> 
> ### Actual result:
> * After the account is removed, the default account is still shifted to the left.
> * After scrolling the last account back, it is shifted to the right.
> 
>  shift.mp4 
> ### Expected result:
> The account positions should be aligned image
> 
> ### Device:
> Pixel 7a, Android 14

@smohamedjavid, @ulisesmac, @OmarBasem do any of you know how to fix that issue? 
It seems related to the flat-list component since it is re-rendering with new data? 
Any ideas for how best to handle it 🤔

**Comment by J-Son89** [comment]
@VolodLytvynenko - I pushed a fix for Issue 1. I will start to look into issue 2 now. will let you know when it's ready for retesting etc 👍

**Comment by ulisesmac** [comment]
> > ## ISSUE 2: [Android] Defalut account is shifted after removing the last account available for removing
> > ### Steps:
> > 
> > 1. Create an additional ETH account.
> > 2. Scroll the accounts to the left.
> > 3. Remove the account.
> > 4. Check the behavior of the last account.
> > 
> > ### Actual result:
> > 
> > * After the account is removed, the default account is still shifted to the left.
> > * After scrolling the last account back, it is shifted to the right.
> > 
> > shift.mp4
> > ### Expected result:
> > The account positions should be aligned image
> > ### Device:
> > Pixel 7a, Android 14
> 
> @smohamedjavid, @ulisesmac, @OmarBasem do any of you know how to fix that issue? It seems related to the flat-list component since it is re-rendering with new data? Any ideas for how best to handle it 🤔

hey @J-Son89 

I think it's due the the scroll position, I'd suggest to recreate the component it instead of rerendering it.

I've faced problems like this in flat lists too often, you had a scroll position, then the size shrinks and the scroll is off.

You could add a react key to this `flat-list` that contains the size of the elements
e.g.:
```clojure
^{:key (str "account-cards-" (count account-cards))}
```
when the size changes, the component will have a different key, so the old one will be dropped and a whole new one will be rendered.

You just need to be careful about the transition from the old to the new component, it shouldn't look remounted/buggy.

Another option might be calling `.scrollToOffset` with `#js{:offset 0}` when we remove the card or when visit the wallet home screen so that we scroll to the beginning, although I haven't tried it.

**Comment by J-Son89** [comment]
> > > ## ISSUE 2: [Android] Defalut account is shifted after removing the last account available for removing
> 
> > > ### Steps:
> 
> > > 
> 
> > > 1. Create an additional ETH account.
> 
> > > 2. Scroll the accounts to the left.
> 
> > > 3. Remove the account.
> 
> > > 4. Check the behavior of the last account.
> 
> > > 
> 
> > > ### Actual result:
> 
> > > 
> 
> > > * After the account is removed, the default account is still shifted to the left.
> 
> > > * After scrolling the last account back, it is shifted to the right.
> 
> > > 
> 
> > > shift.mp4
> 
> > > ### Expected result:
> 
> > > The account positions should be aligned image
> 
> > > ### Device:
> 
> > > Pixel 7a, Android 14
> 
> > 
> 
> > @smohamedjavid, @ulisesmac, @OmarBasem do any of you know how to fix that issue? It seems related to the flat-list component since it is re-rendering with new data? Any ideas for how best to handle it 🤔
> 
> 
> 
> hey @J-Son89 
> 
> 
> 
> I think it's due the the scroll position, I'd suggest to recreate the component it instead of rerendering it.
> 
> 
> 
> I've faced problems like this in flat lists too often, you had a scroll position, then the size shrinks and the scroll is off.
> 
> 
> 
> You could add a react key to this `flat-list` that contains the size of the elements
> 
> e.g.:
> 
> ```clojure
> 
> ^{:key (str "account-cards-" (count account-cards))}
> 
> ```
> 
> when the size changes, the component will have a different key, so the old one will be dropped and a whole new one will be rendered.
> 
> 
> 
> You just need to be careful about the transition from the old to the new component, it shouldn't look remounted/buggy.
> 
> 
> 
> Another option might be calling `.scrollToOffset` with `#js{:offset 0}` when we remove the card or when visit the wallet home screen so that we scroll to the beginning, although I haven't tried it.

Ok thanks Ulises, I'll try those out 🙏

**Comment by J-Son89** [comment]
Hi @VolodLytvynenko - please check this video to see the bugs fixed. let me know what you think.
https://github.com/status-im/status-mobile/assets/22799766/caafe0d6-1f2d-4196-b476-82d530db59a0

**Comment by VolodLytvynenko** [comment]
@J-Son89 thank you for PR. No issues from my side. PR is ready to be merged

**Pull request merged**